src/udev/udevadm-hwdb.c

```diff
@@ -619,9 +639,18 @@ static int adm_hwdb(struct udev *udev, int argc, char *argv[]) {
                 return EXIT_FAILURE;
         }
 
+        if (root) {
```

Comment:
Condition is always true. Maybe `if (strlen(root))`?

Also I think this logic have to go below inside `if (update)`

Reply:
Good catch! Yes, the meaning is "if we passed --root", so that -o test.bin does not turn into -o /test.bin.